nibabel/tests/test_scripts.py

```diff
@@ -54,15 +54,62 @@ def script_test(func):
 
 
 @script_test
-def test_nib_ls():
+def _test_nib_ls_example4d(opts=[], hdrs_str=""):
     # test nib-ls script
     fname = pjoin(DATA_PATH, 'example4d.nii.gz')
     expected_re = (" (int16|[<>]i2) \[128,  96,  24,   2\] "
-                   "2.00x2.00x2.20x2000.00  #exts: 2 sform$")
-    cmd = ['nib-ls', fname]
+                   "2.00x2.00x2.20x2000.00  #exts: 2%s sform$"
+                   % hdrs_str)
+    cmd = ['nib-ls'] + opts + [fname]
     code, stdout, stderr = run_command(cmd)
     assert_equal(fname, stdout[:len(fname)])
-    assert_not_equal(re.match(expected_re, stdout[len(fname):]), None)
+    assert_re_in(expected_re, stdout[len(fname):])
+
+@script_test
+def test_nib_ls():
+    yield _test_nib_ls_example4d
+    yield _test_nib_ls_example4d, ['-H', 'dim,bitpix'], " \[  4 128  96  24   2   1   1   1\] 16"
+
+@script_test
+def test_nib_ls_multiple():
+    # verify that correctly lists/formats for multiple files
+    fnames = [
+        pjoin(DATA_PATH, f)
+        for f in ('example4d.nii.gz', 'example_nifti2.nii.gz', 'small.mnc', 'nifti2.hdr')
+    ]
+    code, stdout, stderr = run_command(['nib-ls'] + fnames)
+    stdout_lines = stdout.split('\n')
+    assert_equal(len(stdout_lines), 4)
+    # they should be indented correctly.  Since all files are int type -
+    ln = max(len(f) for f in fnames)
+    assert_equal([l[ln:ln+2] for l in stdout_lines], [' i']*4,
```

Comment:
is there a convenient way to run check only on changes?  otherwise -- there is more to fix up:
```
$> flake8 nibabel/testing/__init__.py
nibabel/testing/__init__.py:3:1: E266 too many leading '#' for block comment
nibabel/testing/__init__.py:8:1: E266 too many leading '#' for block comment
nibabel/testing/__init__.py:23:5: F401 'assert_false' imported but unused
nibabel/testing/__init__.py:23:5: F401 'assert_not_equal' imported but unused
nibabel/testing/__init__.py:23:5: F401 'assert_raises' imported but unused
nibabel/testing/__init__.py:179:80: E501 line too long (84 > 79 characters)
2 10297 ->1.....................................:Mon 30 Nov 2015 01:34:44 PM EST:.
(git)hopa:~nipy-suite/nibabel[nibls-header]git
$> flake8 nibabel/tests/test_scripts.py
nibabel/tests/test_scripts.py:12:1: F401 're' imported but unused
nibabel/tests/test_scripts.py:21:1: F401 'assert_not_equal' imported but unused
nibabel/tests/test_scripts.py:24:1: F401 'assert_array_equal' imported but unused
nibabel/tests/test_scripts.py:41:16: E251 unexpected spaces around keyword / parameter equals
nibabel/tests/test_scripts.py:41:18: E251 unexpected spaces around keyword / parameter equals
nibabel/tests/test_scripts.py:42:20: E251 unexpected spaces around keyword / parameter equals
nibabel/tests/test_scripts.py:42:22: E251 unexpected spaces around keyword / parameter equals
nibabel/tests/test_scripts.py:51:29: E261 at least two spaces before inline comment
nibabel/tests/test_scripts.py:67:1: E302 expected 2 blank lines, found 1
nibabel/tests/test_scripts.py:70:80: E501 line too long (97 > 79 characters)
nibabel/tests/test_scripts.py:72:1: E302 expected 2 blank lines, found 1
nibabel/tests/test_scripts.py:77:80: E501 line too long (89 > 79 characters)
nibabel/tests/test_scripts.py:105:80: E501 line too long (95 > 79 characters)
nibabel/tests/test_scripts.py:106:80: E501 line too long (95 > 79 characters)
nibabel/tests/test_scripts.py:107:80: E501 line too long (94 > 79 characters)
nibabel/tests/test_scripts.py:108:80: E501 line too long (81 > 79 characters)
nibabel/tests/test_scripts.py:114:1: E303 too many blank lines (3)
```
meanwhile will trim the lines I have introduced